File: mario.py
from cs50 import get_int

# Prompts the user for the pyramid's height
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# Prints the pyramid
for i in range(height):

    # Prints the left spaces
    print(" " * (height - (i + 1)), end="")

    # Prints the left bricks
    for j in range(i + 1):
        print("#", end="")

    # Prints the middle spaces
    print("  ", end="")

    # Prints the right bricks
    for j in range(i + 1):
        print("#", end="")
    print()

File: mario.c
// Creates a pyriamid using hashes (#) as bricks
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // Prompts the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    // Prints a pyramid of that height
    for (int i = n; i > 0; i--)
    {
        // Print row of bricks
        print_row(i - 1, n);
    }
}

void print_row(int spaces, int bricks)
{
    // Prints the spaces
    for (int k = spaces; k > 0; k--)
    {
        printf(" ");
    }

    // Prints the right-aligned pyriamid bricks
    for (int l = bricks; spaces < l; l--)
    {
        printf("#");
    }

    printf("  ");

    // Prints the left-aligned pyriamid bricks
    for (int m = spaces; m < bricks; m++)
    {
        printf("#");
    }
    printf("\n");
}
